protocols/kad/src/protocol.rs

```diff
@@ -105,10 +106,13 @@ impl TryFrom<proto::message::Peer> for KadPeer {
 
         let mut addrs = Vec::with_capacity(peer.addrs.len());
         for addr in peer.addrs.into_iter() {
-            let as_ma = Multiaddr::try_from(addr).map_err(invalid_data)?;
-            addrs.push(as_ma);
+            match Multiaddr::try_from(addr) {
+                Ok(a) => addrs.push(a),
+                Err(e) => {
+                    debug!("Unable to parse multiaddr: {e:?}");
```

Comment:
```suggestion
                    debug!("Unable to parse multiaddr: {e}");
```

I'd prefer display over debug.